pkg/datarepair/checker/config.go

```diff
@@ -45,11 +47,32 @@ func (c Config) initialize(ctx context.Context) (Checker, error) {
 		return nil, errs.New("unable to get master db instance")
 	}
 	o := overlay.LoadServerFromContext(ctx)
-	redisQ, err := redis.NewQueueFrom(c.QueueAddress)
+
+	driver, _, err := utils.SplitDBURL(c.QueueAddress)
 	if err != nil {
-		return nil, Error.Wrap(err)
+		return nil, err
+	}
+
+	var repairQueue *queue.Queue
+	switch driver {
+	case "postgres":
+		postgresDB, ok := ctx.Value("masterdb").(interface {
+			RepairQueueDB() storage.Queue
+		})
+		if !ok {
+			return nil, errs.New("unable to get master db instance")
+		}
+		repairQueue = queue.NewQueue(postgresDB.RepairQueueDB())
```

Comment:
It should return a compatible queue already, so the `queue.NewQueue` shouldn't be necessary here.